sim/core/spell_outcome.go

```diff
@@ -468,14 +470,10 @@ func (spell *Spell) outcomeRangedHitAndCrit(sim *Simulation, result *SpellResult
 	chance := 0.0
 
 	if spell.Unit.PseudoStats.InFrontOfTarget {
-		if !result.applyAttackTableMissNoDWPenalty(spell, attackTable, roll, &chance) {
-			if result.applyAttackTableCritSeparateRoll(sim, spell, attackTable, countHits) {
-				result.applyAttackTableBlock(spell, attackTable, roll, &chance)
-			} else {
-				if !result.applyAttackTableBlock(spell, attackTable, roll, &chance) {
-					result.applyAttackTableHit(spell, countHits)
-				}
-			}
+		if !result.applyAttackTableMissNoDWPenalty(spell, attackTable, roll, &chance) &&
+			!result.applyAttackTableDodge(spell, attackTable, roll, &chance) &&
+			!result.applyAttackTableCritSeparateRoll(sim, spell, attackTable, countHits) {
+			result.applyAttackTableHit(spell, countHits)
 		}
 	} else {
 		if !result.applyAttackTableMissNoDWPenalty(spell, attackTable, roll, &chance) &&
```

Comment:
You also need to handle the Dodge outcome in the case where the Hunter is attacking from behind. In fact, now that Blocks are no longer possible for ranged attacks, I don't think you even need to branch off `InFrontOfTarget` anymore - it should be the same attack table in both cases.